extensions/validation/src/main/java/org/mule/extension/validation/api/ObjectSource.java

```diff
@@ -48,27 +48,18 @@
 
   @Parameter
   @Optional
-  private String ref;
+  @XmlHints(allowInlineDefinition = false)
+  private T ref;
 
   public ObjectSource() {}
 
-  public ObjectSource(String type, String ref) {
+  public ObjectSource(String type, T ref) {
     this.type = type;
     this.ref = ref;
   }
 
-  public final T getObject(MuleContext muleContext) {
-    boolean hasType = !StringUtils.isBlank(type);
-    boolean hasRef = !StringUtils.isBlank(ref);
-
-    checkArgument(!(hasType && hasRef), "type and ref attributes are mutually exclusive. Please provide only one of them");
-    checkArgument(hasType ^ hasRef, "One of class or ref attributes are required. Please provide one of them");
-
-    if (hasRef) {
-      return doGetByRef(muleContext);
-    } else {
-      return doGetByClassName();
-    }
+  public final T getObject() {
+    return !isBlank(type) ? doGetByClassName() : ref;
```

Comment:
doGetByClassName() should cache the object it returns and not return a new one each time. The getRegistry().applyProcessors() thing should be one in here one the instance is first created and not all the time it is used. That means that doGetByClassName should be synchronized and that this method should receive the muleContext